Python 2/3 compatibility in mock_services

vvolkl opened the pull request:
Only touches make_repo.py script, the other mock services should be rewritten in plain C. (See https://github.com/cvmfs/cvmfs/issues/3068 - will be done in a future PR).
Just  ran 2to3 and did some small improvements: Replaced tabs with spaces, used `random.getrandbytes` where possible. The updated call to get random bytes also speeds up the script quite a bit: Down to <1s from 6s for `make_repo.py -d 2 -n 2 /tmp/testfs`.

vvolkl commented:
@cernvm-bot unittest

jblomer commented:
@cernvm-bot help

jblomer reviewed (approved):
Looks good!  The only way to test this is through test 525, which does not run by default (not in the quick suite).  I'll trigger a full integration test run.

jblomer commented:
@cernvm-bot cloudtest full

jblomer commented:
We still seem to have a problem on Debian 11: https://lcgapp-services.cern.ch/cvmfs-jenkins/job/CvmfsCloudTesting/1521/CVMFS_PLATFORM=debian11,CVMFS_PLATFORM_CONFIG=x86_64,label=trampoline/testReport/junit/(root)/ServerIntegrationTests/Split_a_Huge_Repository_into_Nested_Catalogs/

(also Ubuntu 20. Possibly more platforms.)

vvolkl commented:
At first glance it just seems like I overlooked that `make_repo.py` may write empty files, and getrandbits throws an exception when requesting 0 bits - but I'll take a closer look! I'll also rebase to put the whitespace changes in a separate commit.

vvolkl committed:
[test] fix whitespace in mock service script

vvolkl committed:
[test] run python 2to3 on mock service script

vvolkl committed:
[test] python 2/3 compatibility in mock service script

Only touches make_repo.py script, the other mock services should be rewritten in plain C. (See #3068).
Just ran 2to3 and did some small improvements: Replaced tabs with spaces, used random.getrandbytes where possible. The updated call to get random bytes also speeds up the script quite a bit: Down to <1s from 6s for make_repo.py -d 2 -n 2 /tmp/testfs.

vvolkl commented:
@cernvm-bot cloudtest full

vvolkl commented:
@cernvm-bot cloudtest full
